publish/test/test.js

```diff
@@ -2,13 +2,13 @@ import { publish } from '../index.js'
 import assert from 'node:assert'
 import { CID } from 'multiformats/cid'
 import pg from 'pg'
-import * as telemetry from '../lib/telemetry.js'
+import * as telemetry from '../../common/telemetry.js'
 
 import { assertApproximately } from '../../test-helpers/assert.js'
 
 const { DATABASE_URL } = process.env
 
-after(telemetry.close)
+after(async () => { await telemetry.publishWriteClient.close() })
```

Comment:
IIRC, we are calling `close` to prevent the write client from blocking the process from exiting.

Ideally, tests should not write any telemetry, they should mock the `recordTelemetryFn`, as we already do in other tests.

Alternatively, the `telemetry` module should implement a `close` function to close all writers.

For example (may not work out of the box):

```js
// usage here - no change
after(telemetry.close)

// implementation in common/telemetry
export const close = async () => {
  await Promise.all([
    publishWriteClient.close(), 
    apiWriterClient.close(), 
    //etc.
  ])
}
```

Reply:
This closing logic was already present I think, this change is just to bring it in line with the new telemetry file.

If we want to change this, maybe it's better scoped in a different PR?

Reply:
I see your point.

The problem I am trying to prevent is that with your changes in place, the test suite does not exit cleanly, because there are background active writers.

Before your change, when executing the tests in `publish/test/test.js`, only one writer was created and this writer was closed by this `after` line. 

With your changes in place, executing this test file creates more than on writer, but we close only one of them. As a result, the test suite never exits.

I confirmed this by running the following command:

```
npm t -w api -- -i -g "getRoundStartEpoch|startRoundTracker"
```

We want to rework this test file anyway; it should not write any telemetry, as I mentioned before. Essentially, we want to port https://github.com/filecoin-station/voyager-api/pull/22 to this repo.

I am going to make two smaller changes: fix this test file to use a mocked telemetry recorder and introduce `common/typings.d.ts` file as discussed in https://github.com/filecoin-station/spark-api/pull/339#discussion_r1652173812

Reply:
See https://github.com/filecoin-station/spark-api/pull/354

In the end, I did not need `common/typings.d.ts`, so I am leaving that change for you to make in this pull request.

Reply:
I think we can remove `telemetry.close()` altogether now, including the implementation.

Reply:
```suggestion
```